Add Micrometer to docs (#1545)
kiegroup/optaplanner: optaplanner-docs/src/main/asciidoc/PlannerConfiguration/PlannerConfiguration-chapter.adoc

----
====

[[monitoring]]
=== Monitoring

> OptaPlanner exposes metrics via https://micrometer.io/[Micrometer] which can be used for monitoring the Solver. OptaPlanner automatically connect to configured registries when used in Quarkus or Spring Boot. In plain Java, add the registry to the global Metrics object:

Reviewer: ```suggestion
OptaPlanner exposes metrics via https://micrometer.io/[Micrometer] which can be used for monitoring the Solver. OptaPlanner automatically connects to configured registries when used in Quarkus or Spring Boot. In plain Java, add the registry to the global registry:
```

[source,java,nowrap]
----
Metrics.addRegistry(new SimpleMeterRegistry());
----

The following metrics are exposed (format may differ depending on registry):

- `optaplanner.solver.errors`: Count the total number of errors that occurred while solving

- `optaplanner.solver.solve-length.active-count`: The number of solvers currently active

> - `optaplanner.solver.solve-length.seconds-max`: How long the oldest active solver been running

Reviewer: ```suggestion
- `optaplanner.solver.solve-length.seconds-max`: Run time of the longest-running currently active solver
```

> - `optaplanner.solver.solve-length.seconds-duration-sum`: The sum of each active solver solve length (ex: if there are two active solvers, one been running for 3 minutes, the other for 1 minute, this would be 4 minutes).

Reviewer: ```suggestion
- `optaplanner.solver.solve-length.seconds-duration-sum`: The sum of each active solver solve length (ex: if there are two active solvers, one running for 3 minutes and the other for 1 minute, this would be 4 minutes).
```
